sycl/include/CL/sycl/accessor2.hpp

```diff
@@ -333,7 +333,7 @@ class accessor :
 public:
   using value_type = DataT;
   using reference = DataT &;
-  using const_reference = const reference;
```

Comment:
Could you clarify what is wrong with the definition in the log message, please?

Reply:
The error message says that 'const_reference' is 'int &', not 'const int &' and I have no idea why.
This fix though helps to eliminate the error.

Reply:
I think it get this.
Current implementation tries to apply const qualifier to the **reference**, which doesn't make sense.
New implementation defines a reference to the `const int` type.

Reply:
I still do not understand why the previous code does not work. 'const reference' adds const qualifier to 'reference' and 'reference' is synonym to 'DataT &'. So, I would think that const add the qualifier to 'DataT &'.

In general, you agree with the fix, right?

Reply:
"<const T> &" vs "const <T&>"

> In general, you agree with the fix, right?

Yes.